### DISCUSSION TITLE
Introduction and Appreciation

### DISCUSSION THREAD

**Discussion by tomstix** [Open-Agriculture/AgIsoStack-plus-plus#89] [General]
Hi, I'm Tom, a Mechanical Engineering student from Germany. I stumbled across this project trying to read into IsoAgLib (turned out to be a little frustrating). So I am super happy that there is put so much effort in developing a modern and well-documented alternative.
I like how the Open Source Farming community is growing in the past years and I would be happy to use my abilities in C++ to contribute here.
I will also share this among the people I know who are interested in Open Source ISOBUS development (there are quite a few, I think) to hopefully get some good attention for this project.

**Comment by ad3154** [comment]
Hello Tom! 
Thanks for your interest in the project. Actually I recently traveled to Germany and it was extremely enjoyable! Farming innovation is very strong there. Our CAN stack project here has a small but growing community of people, and is still young, but is progressing quickly! It's a passion project, and I'm sure it has its issues, but I hope that some day it becomes the go-to starting point in the open source farming community (and maybe for some companies also, haha) for ISOBUS and J1939 applications at least for C++. That is of course a lofty goal, and there are other, institutional barriers as well, such as AEF membership being quite expensive and apparently being restricted to companies, but it's a dream, for sure.

As such, we would welcome your contributions and/or some additional visibility. 

I am working on drafting up a contributing document to help get new contributors going that outlines the repo's style as it currently is, some instructions to set up a development environment, and some documentation for the internal structure of the stack to make contributing easier.

Anyways, it's always very exciting to see that there's other people out there in the open source farming community. It turns out the community is larger than I thought!

Thanks,
Adrian
